### DISCUSSION TITLE
Units for Neo-Hookean Parameters

### DISCUSSION THREAD

**Discussion by florianHofherr** [bulletphysics/bullet3#3443] [General]
Hi everyone,

I want to do a deformable simulation in PyBullet using a Neo-Hookean model with known and "real-world-interpretable" physical parameters. In the examples provided in the repo, the parameters _mu_ and _lambda_ used are in the range 400 - 600, for example. This made me wonder in which units they actually are. I assume, that they are the Lamé parameters, i.e.

lambda = E*nu/(1+nu)/(1-2*nu)
mu = E /2/(1+nu)

with the Young's modulus _E_ and the poisson ratio _nu_. Since _E_ is usually in GPa and _nu_ is in [0, 0.5], _mu_ and _lambda_ should be in the range of 1e9 Pa. The values described above would indicate that maybe _mu_ and _lambda_ are given in MPa, but this is just me guessing.  

Are the units that are expected documented somewhere? Also, are mass in kg, and length in m? Since this will affect the density, this probably has a big impact on the simulation as well

Thank you in advance!

**Comment by erwincoumans** [comment]
Yes, mass is in kg and length in meter.
I would expect the Young modulus units to be N/m2, one of the examples uses the range [0,2000], but it depends on other parameters (see [demo](https://github.com/bulletphysics/bullet3/blob/master/examples/DeformableDemo/VolumetricDeformable.cpp#L246)).
poisson ration has no units, and in the range [0,0.5]

**Comment by mannanabdul009** [comment]
Why did you set 0.5?
